docs/design/2018-10-24-view-support.md

```diff
@@ -5,65 +5,131 @@
+* TableInfo.Columns  
+    `TableInfo.Columns` only stores view's column origin names.
+ 
 ## Rationale
-`VIEW` is just a tableinfo object store with a `SELECT` statement, and only supports two types of DDL operations, which are "`CREATE OR REPLACE VIEW`" and "`DROP`". Currently, view can only be used within the `SELECT` statement.  
-Let me describe more details about the view DDL operation:
-1. Create a VIEW  
-  1.1 Parse the create view statement and build a logical plan for select cause part. If any grammar error occurs, return errors to parser.   
-  1.2 Examine view definer's privileges. Definer should own both `CREATE_VIEW` and base table's `SELECT` privileges.  
-  1.3 If `ViewFieldList` cause is empty, then we should generate view column name from `SelectStmt` cause.  
-  1.4 Replace any wildcard with column name in `SELECT` cause.  
-  1.5 Test `SELECT` cause with some rules to see if this view is updateable.  
-  1.6 Store the view definition into following tables
-     * `INFORMATION_SCHEMA.VIEW_COLUMN_USAGE` lists one row for each column in a view including the base table of the column where possible 
-     * `INFORMATION_SCHEMA.VIEW_TABLE_USAGE` lists one row for each table used in a view 
-     * `INFORMATION_SCHEMA.VIEWS` lists one row for each view
-      
-2. Drop a view  
-  Implement DROP VIEW grammar, and delete the existing view tableinfo object.  
-3. Select from a view  
-  3.1 In function `func (b *PlanBuilder) buildDataSource(tn *ast.TableName) (LogicalPlan, error)`, if `tn *ast.TableName` is a view, then we build a select logicalplan from view's view_select string.   
-4. Insert/Update into a view  
-  I do not have any idea now, and will implement later.   
-5. Describe a view  
-  Generate the view select logicalplan. If any error occurs, return the error to parser. Otherwise, return `SELECT` cause's column info.
+1. Create VIEW  
+   This proposal only supports the following grammar to create view:
+   ```mysql
+    CREATE
+        [OR REPLACE]
+        [ALGORITHM = {UNDEFINED | MERGE | TEMPTABLE}]
+        [DEFINER = { user | CURRENT_USER }]
+        [SQL SECURITY { DEFINER | INVOKER }]
+        VIEW view_name [(column_list)]
+        AS select_statement
+        [WITH [CASCADED | LOCAL] CHECK OPTION]
+    ```
+    1. Parse the create view statement and build a logical plan for select clause part. If any grammar error occurs, return errors to parser.   
+    2. Examine view definer's privileges. Definer should own both `CREATE_VIEW_PRIV` privilege and base table's `SELECT` privilege. We will reuse `CREATE_PRIV` privilege when implement `CREATE VIEW` feature and will support `CREATE_VIEW_PRIV` check later.  
+    3. Examine create view statement. If the ViewFieldList clause part is empty, then we should generate view column names from SelectStmt clause. Otherwise check len(ViewFieldList) == len(Columns from SelectStmt). And then we save column names to `TableInfo.Columns`.
```

Comment:
```suggestion
    3. Examine create view statement. If the `ViewFieldList` clause part is empty, then we should generate view column names from the `SelectStmt` clause. Otherwise check `len(ViewFieldList) == len(Columns from SelectStmt)`. And then we save column names to `TableInfo.Columns`.
```